sympy/parsing/sympy_parser.py

```diff
@@ -931,6 +931,11 @@ def visit_BinOp(self, node):
         if node.op.__class__ in self.operators:
             sympy_class = self.operators[node.op.__class__]
             right = self.visit(node.right)
+            left = self.visit(node.left)
+            if isinstance(node.left, ast.UnaryOp) and (isinstance(node.right, ast.UnaryOp) == 0) and sympy_class in ('Mul','Pow'):
+                temp = right
+                right = left
+                left = temp
```

Comment:
There should be no need for ``temp``. Just assign ``left, right = right, left``.